crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -147,6 +161,329 @@ async fn generate_cac(
+        }
+    }
+
+    contexts_overrides_values.sort_by(|a, b| {
+        (validate_dimensions_and_calculate_priority(
+            "context",
+            &(b.0).condition,
+            dimension_schema_map,
+        )
+        .unwrap())
+        .cmp(
+            &(validate_dimensions_and_calculate_priority(
+                "context",
+                &(a.0).condition,
+                dimension_schema_map,
+            )
+            .unwrap()),
+        )
+    });
+
+    let resolved_dimensions = reduce(contexts_overrides_values, default_config_val)?;
+    for rd in resolved_dimensions {
+        match (
+            rd.get("can_be_reduced"),
+            rd.get("req_payload"),
+            rd.get("req_payload").and_then(|v| v.get("id")),
+            rd.get("req_payload").and_then(|v| v.get("override_id")),
+            rd.get("req_payload").and_then(|v| v.get("to_be_deleted")),
+            rd.get("req_payload").and_then(|v| v.get("override")),
+        ) {
+            (
+                Some(Value::Bool(true)),
+                Some(Value::Object(request_payload)),
+                Some(Value::String(cid)),
+                Some(Value::String(oid)),
+                Some(Value::Bool(to_be_deleted)),
+                Some(override_val),
+            ) => {
+                if *to_be_deleted {
+                    if is_approve {
+                        let _ = delete_context_api(cid.clone(), user.clone(), conn).await;
+                    }
+                    og_contexts.retain(|x| x.id != *cid);
+                } else {
+                    if is_approve {
+                        let _ = delete_context_api(cid.clone(), user.clone(), conn).await;
+                        let put_req = construct_new_payload(request_payload);
+                        let _ = put(put_req, conn, false, &user);
+                    }
+
+                    let new_id = hash(override_val);
+                    og_overrides.insert(new_id.clone(), override_val.clone());
+
+                    let mut ctx_index = 0;
+                    let mut delete_old_oid = true;
+
+                    for (ind, ctx) in og_contexts.iter().enumerate() {
+                        if ctx.id == *cid {
+                            ctx_index = ind;
+                        } else if ctx.override_with_keys.contains(oid) {
+                            delete_old_oid = false;
+                        }
+                    }
+
+                    let mut elem = og_contexts[ctx_index].clone();
+                    elem.override_with_keys = [new_id];
+                    og_contexts[ctx_index] = elem;
+
+                    if delete_old_oid {
+                        og_overrides.remove(oid);
+                    }
+                }
+            }
+            _ => continue,
+        }
+    }
+
+    Ok(Config {
+        contexts: og_contexts,
+        overrides: og_overrides,
+        default_configs: default_config,
+    })
+}
+
+#[put("/reduce")]
+async fn reduce_context(
```

Comment:
@Sauravcv98 is there any specific reason we kept this api under config and not context ?

Reply:
Hey this actually reduces the entire Config, even the response of this is of type Config

Reply:
then probably its name should be reduce_config @knutties @Datron  ?

Reply:
Yeah true, was thinking the same
Will rename this to reduce_config